### ISSUE TITLE
locationTemplate config property not working correctly

### ISSUE THREAD

**Issue by patodevilla** [transistorsoft/cordova-background-geolocation-lt#730]
## Your Environment
* 2.11.0
* Platform: Android
* 6.3.0
* Lenovo Phablet 2
* Cordova version 7.1.0
* Cordova platform version android 6.3.0
* Plugin config

    {

      // *******************
      // GEOLOCATION OPTIONS
      // *******************

      // Common Options
      desiredAccuracy: 0, // default 0 options: 0, 10, 100, 1000 (0-> HIGHEST POWER, HIGHEST ACCURACY)
      distanceFilter: 5, // default 10 Min distance (meters) to move horizontally before an update event

      // Android Options

      allowIdenticalLocations: ENV.DEBUG_LOCATION, // default false

      // ****************************
      // ACTIVITY RECOGNITION OPTIONS
      // ****************************

      // Common Options
      activityRecognitionInterval: 5000, //Larger values will improve battery life. default: 10000
      minimumActivityRecognitionConfidence: 50, //default: 75 (percentaje)

      // *********************
      // HTTP / SQLite config
      // *********************

      url: url,
      httpTimeout: 4000, //default 60000
      headers: {
        'Authorization': authorization
      },
      method: "POST",
      httpRootProperty: '.',
      locationTemplate: '{ "smartphone_device_logs": { "payload": { "location": { "lat": <%= latitude %>, "lon": <%= longitude %> }, "event": "<%= event %>", "activity_type": "<%= activity.type %>", "accuracy": "<%= accuracy %>", "altitude_accuracy": "<%= altitude_accuracy %>", "altitude": "<%= altitude %>" } } }',

      // *******************
      // APPLICATION OPTIONS
      // *******************

      // Common Options
      stopOnTerminate: false, //default true
      startOnBoot: true, //default false

      // Android Options
      foregroundService: true, //default false
      notificationTitle: "Prysmex",
      notificationText: "Location service activated",
      notificationSmallIcon: true,
      notificationLargeIcon: true, //default undefined

      // ***********************
      // LOGGING & DEBUG OPTIONS
      // ***********************

      debug: true
    }

## Expected Behavior

The template should structure the json request

## Actual Behavior

I am getting this in my server (it does not match the locationTemplate)

<ActionController::Parameters {"event"=>"geofence", "is_moving"=>true, "uuid"=>"ed056446-84fb-4159-acc0-63d9b2968382", "timestamp"=>"2018-07-17T20:54:01.207Z", "odometer"=>1.1, "mock"=>true, "coords"=>{"latitude"=>25.69855236816227, "longitude"=>-100.29797567551513, "accuracy"=>5.5, "speed"=>0.07, "heading"=>-1, "altitude"=>42.5}, "activity"=>{"type"=>"on_foot", "confidence"=>98}, "battery"=>{"is_charging"=>false, "level"=>1}, "geofence"=>{"identifier"=>"Planta Monterrey", "action"=>"ENTER", "extras"=>{"project_id"=>"1"}}, "extras"=>{}, "format"=>"json", "locale"=>"es", "controller"=>"api/v1/smartphone_device_logs", "action"=>"create", "smartphone_device_log"=>{}} permitted: false>

**Comment by christocracy** [comment]
Read carefully the docs for the #ready method.  Take special note of how the #ready method consumes the supplied config {} only on the first install of the app.

Use reset: true during development when you’re modifying the config.

**Comment by patodevilla** [comment]
That was quick! Thank you very much, I'll try that

**Comment by christocracy** [comment]
You get what you pay for:  prompt support.

**Comment by christocracy** [comment]
Btw, customers should use the private repo.

**Comment by patodevilla** [comment]
Ok, I'll use the private repo the next time! 

I finally found my error, It appears that I had over 150 locations on the device. I ran destroyLocations() and everything seems to be ok now.

**Comment by patodevilla** [comment]
@christocracy I am still having trouble with my `locationTemplate`. When I initiate my app, all the locations are sent with what I think is the default `locationTemplate`, and for some reason when I call `destroyLocations()` the structure seems to get fixed. The problem is that the next time i run `ember cdv:proxy run android --device` the same problem araises.

I added the `reset: true` option to my `ready` function options. 

I also checked the `locationTemplate` with te `getState` function and it appears to be correctly configured. 

Am I missing something?
